gammapy/cube/fit.py

```diff
@@ -804,7 +811,7 @@ def to_dict(self, filename=""):
             "name": self.name,
             "type": self.tag,
             "likelihood": self.likelihood_type,
-            "models": [_.name for _ in self.model],
+            "models": [_.name for _ in self.models],
```

Comment:
This fail for the None case no ?

Reply:
Yes.
How should we fix?
Adapt this and other callers and add an if/else?
Or create `self._models = SkyModels([])` in `@models.setter`?

Reply:
we have this in the spectrumDatasetOnOff
"""
        if self.model is not None:
            models = [_.name for _ in self.model]
        else:
            models = []

        return {
            "name": self.name,
            "type": self.tag,
            "models": models,
            "likelihood": self.likelihood_type,
            "filename": filename,
        }
"""

Reply:
So I put the `if` and we allow `None`?
I'm not sure, but I think that would also be my preference.

Reply:
I've put the same if in both places. @QRemy - Thanks for the advice!